# **Tarea Nº 3**
Scope: If, Loops, Functions

Curso:          Fundamentos de Programación en Python para Macroeconomía y Finanzas     
Profesor:       Victor Fuentes Campos     

- Fecha y hora de entrega: Miércoles 10 de enero, 11:59am
- Agregue los nombres de las personas con las que discutió esta tarea: ____ Piero Fernández Dávila - Rafael Vilca ______
- Envíe su tarea haciendo el `push` de su código a su repo en GitHub Classroom: 

As of 2022 Q2, there are almost 3.5 million Andriod apps on Google Play and 1.6 million iOS apps on the App Store, and counting. Giving the digital revolution, there is an increase interest in investment opportunities in this sector

<img src="popular_app_stores.png" max-width="300">

Under this context, you will assess a financial committee interested in investing in software development. The goal is to determine the kinds of apps that are likely to attract more users because the number of people using our apps affect our revenue. For this purpose, you will access to data from Google Play and App Store.

- Google Play: you can dowload the dataset containing approximately 11k Android apps from [this link](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv). It was collected in August 2018
- App Store: you can dowload the dataset containing roughly 7k iOS apps from [this link](https://dq-content.s3.amazonaws.com/350/AppleStore.csv). It was collected in July 2017

## Exploring data

1. Open the two datasets we mentioned above, and save both as lists of lists

In [1]:
# Solution
import csv

# Definimos una función para leer el csv y pasar a lista de listas
def read_csv_file(file_name):
    data = []
    with open(file_name, 'r', encoding='utf-8') as file:
        csv_reader = csv.reader(file)
        for row in csv_reader:
            data.append(row)
    return data

# Damos los nombres
google_play_file = 'googleplaystore.csv'
apple_store_file = 'AppleStore.csv'

# Los guardamos como lista de listas 
google_play_data = read_csv_file(google_play_file)
apple_store_data = read_csv_file(apple_store_file)


2. Print the first few rows of each dataset and find the number of rows and columns of each dataset

In [2]:
# Solution
for fila in google_play_data[:5]:
    print(fila)
print("\n")
for fila1 in apple_store_data[:5]:
    print(fila1)
print("\n")
num_filas = len(google_play_data)
num_columnas = len(google_play_data[0])

print(num_filas)
print(num_columnas)

print("\n")
num_filas1 = len(apple_store_data)
num_columnas1 = len(apple_store_data[0])

print(num_filas1)
print(num_columnas1)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']
['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']
['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']
['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_ra

For more info about the columns in the App Store dataset, visit [this link](https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps?select=AppleStore.csv)

## Preparing data

3. As noted before in the [discussion forum](https://www.kaggle.com/datasets/lava18/google-play-store-apps/discussion/66015), there is an error in the Google Play dataset. Please, remove the entry 10472 and explain shortly why

In [22]:
# Solution
if len(google_play_data) > 10472:
    del google_play_data[10472]

Google_play_data = len(google_play_data)
print("Los datos de la columna de categoria para la fila 10472 no esta registrada, por lo que los datos se corrieron una columna hacia la izquierda")
print(Google_play_data)

Los datos de la columna de categoria para la fila 10472 no esta registrada, por lo que los datos se corrieron una columna hacia la izquierda
10840


4. Count the number of duplicates in the Google Play dataset. You must build a function to loop through the Google Play dataset

In [23]:
# Solution
def count_number_duplicates(google_play_data):
    apps_unicas = []
    apps_duplicadas = []
    
    for fila in google_play_data[1:]:
        nombre = fila[0]
        if nombre in apps_unicas:
            apps_duplicadas.append(nombre)
        else:
            apps_unicas.append(nombre)
    num_duplicadas = len (apps_duplicadas)
    return num_duplicadas
numero_duplicadas = count_number_duplicates(google_play_data)
numero_duplicadas

1181

In [24]:
# remove duplicates
def quitar_duplicadas(google_play_data):
    revisadas = set()
    apps_unicas = []
    
    apps_unicas.append(google_play_data[0])
    
    for fila in google_play_data[1:]:
        nombres = fila[0]
        if nombres not in revisadas:
            apps_unicas.append(fila)
            revisadas.add(nombres)
    return apps_unicas
google_play_data1 = quitar_duplicadas(google_play_data)
print(len(google_play_data1))

9659


5. a. Create a dictionary, where each dictionary key is a unique app name and the corresponding dictionary value is the highest number of reviews of that app


In [25]:
# Solution
def crear_diccionario(google_play_data1):
    diccionario = {}
    
    for fila in google_play_data1[1:]:
        nombres_app = fila[0]
        num_reviews = fila[3] 
        
        if nombres_app in diccionario:
            if num_reviews > diccionario[nombres_app]:
                diccionario[nombres_app] = num_reviews
        else:
            diccionario[nombres_app] = num_reviews
    return diccionario

nuevo_diccionario = crear_diccionario(google_play_data1)

print(nuevo_diccionario)
    
    

{'Photo Editor & Candy Camera & Grid & ScrapBook': '159', 'Coloring book moana': '967', 'U Launcher Lite – FREE Live Cool Themes, Hide Apps': '87510', 'Sketch - Draw & Paint': '215644', 'Pixel Draw - Number Art Coloring Book': '967', 'Paper flowers instructions': '167', 'Smoke Effect Photo Maker - Smoke Editor': '178', 'Infinite Painter': '36815', 'Garden Coloring Book': '13791', 'Kids Paint Free - Drawing Fun': '121', 'Text on Photo - Fonteee': '13880', 'Name Art Photo Editor - Focus n Filters': '8788', 'Tattoo Name On My Photo Editor': '44829', 'Mandala Coloring Book': '4326', '3D Color Pixel by Number - Sandbox Art Coloring': '1518', 'Learn To Draw Kawaii Characters': '55', 'Photo Designer - Write your name with shapes': '3632', '350 Diy Room Decor Ideas': '27', 'FlipaClip - Cartoon animation': '194216', 'ibis Paint X': '224399', 'Logo Maker - Small Business': '450', "Boys Photo Editor - Six Pack & Men's Suit": '654', 'Superheroes Wallpapers | 4K Backgrounds': '7699', 'Mcqueen Color

5. b. Use the information stored in the dictionary and create a new dataset, which will have only one entry per app (and for each app, we'll only select the entry with the highest number of reviews).


In [26]:
def crear_dataset(google_play_data1, nuevo_diccionario):
    nuevo_google_play_data = [google_play_data1[0]]

    for fila in google_play_data1[1:]:
        nombres_app = fila[0]
        num_reviews = fila[3]

        if num_reviews == nuevo_diccionario[nombres_app]:
            nuevo_google_play_data.append(fila)
    return nuevo_google_play_data

nuevo_google_play_data = crear_dataset(google_play_data1, nuevo_diccionario)

nuevo_google_play_data

[['App',
  'Category',
  'Rating',
  'Reviews',
  'Size',
  'Installs',
  'Type',
  'Price',
  'Content Rating',
  'Genres',
  'Last Updated',
  'Current Ver',
  'Android Ver'],
 ['Photo Editor & Candy Camera & Grid & ScrapBook',
  'ART_AND_DESIGN',
  '4.1',
  '159',
  '19M',
  '10,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design',
  'January 7, 2018',
  '1.0.0',
  '4.0.3 and up'],
 ['Coloring book moana',
  'ART_AND_DESIGN',
  '3.9',
  '967',
  '14M',
  '500,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design;Pretend Play',
  'January 15, 2018',
  '2.0.0',
  '4.0.3 and up'],
 ['U Launcher Lite – FREE Live Cool Themes, Hide Apps',
  'ART_AND_DESIGN',
  '4.7',
  '87510',
  '8.7M',
  '5,000,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design',
  'August 1, 2018',
  '1.2.4',
  '4.0.3 and up'],
 ['Sketch - Draw & Paint',
  'ART_AND_DESIGN',
  '4.5',
  '215644',
  '25M',
  '50,000,000+',
  'Free',
  '0',
  'Teen',
  'Art & Design',
  'June 8, 2018',
  'Varies with device',
  '4.2 a

5. c. Explore the cleaned dataset, it should have 9,659 rows

In [27]:
# Solution
print(len(nuevo_google_play_data))

9659


6. Remove non-english apps from both Android and Apple datasets. You should write a function that takes a string, if there's more than three characters that fall outside the ASCII range (0 - 127), then the function should return False (identify the string as non-English), otherwise it should return True. Hint: use the [```ord()```  function](https://www.w3schools.com/python/ref_func_ord.asp)

In [13]:
# For instance
correct_english_name = "Instachat 😜"  # 1 character outside the ASCII range (😜)
incorrect_english_name = "爱奇艺PPS -《欢乐颂2》电视剧热播" # 13 character outside the ASCII range (ie 播)

In [14]:
print(ord("I"))
print(ord("t"))
print(ord("-"))
print(ord("😜"))
print(ord("爱"))

73
116
45
128540
29233


In [29]:
# Solution
def es_ingles(string):
    no_ascii = 0
    
    for caracter in string:
        if ord(caracter)>127:
            no_ascii +=1
            
        if no_ascii >3:
            return False
    return True
def filtro_apps_ingles(nuevo_google_play_data, name_index):
    apps_ingles = []
    
    for fila in nuevo_google_play_data[1:]:
        nombres_app = fila[name_index]
        
        if es_ingles(nombres_app):
            apps_ingles.append(fila)
    apps_ingles.insert(0, nuevo_google_play_data[0])
    return apps_ingles

google_play_ingles = filtro_apps_ingles(nuevo_google_play_data, 0)
google_play_ingles

[['App',
  'Category',
  'Rating',
  'Reviews',
  'Size',
  'Installs',
  'Type',
  'Price',
  'Content Rating',
  'Genres',
  'Last Updated',
  'Current Ver',
  'Android Ver'],
 ['Photo Editor & Candy Camera & Grid & ScrapBook',
  'ART_AND_DESIGN',
  '4.1',
  '159',
  '19M',
  '10,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design',
  'January 7, 2018',
  '1.0.0',
  '4.0.3 and up'],
 ['Coloring book moana',
  'ART_AND_DESIGN',
  '3.9',
  '967',
  '14M',
  '500,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design;Pretend Play',
  'January 15, 2018',
  '2.0.0',
  '4.0.3 and up'],
 ['U Launcher Lite – FREE Live Cool Themes, Hide Apps',
  'ART_AND_DESIGN',
  '4.7',
  '87510',
  '8.7M',
  '5,000,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design',
  'August 1, 2018',
  '1.2.4',
  '4.0.3 and up'],
 ['Sketch - Draw & Paint',
  'ART_AND_DESIGN',
  '4.5',
  '215644',
  '25M',
  '50,000,000+',
  'Free',
  '0',
  'Teen',
  'Art & Design',
  'June 8, 2018',
  'Varies with device',
  '4.2 a

In [30]:
# Solution
def es_ingles(string):
    no_ascii = 0
    
    for caracter in string:
        if ord(caracter)>127:
            no_ascii +=1
            
        if no_ascii >3:
            return False
    return True
def filtro_apps_ingles(apple_store_data, name_index):
    apps_ingles = []
    
    for fila in apple_store_data[1:]:
        nombres_app = fila[name_index]
        
        if es_ingles(nombres_app):
            apps_ingles.append(fila)
    apps_ingles.insert(0, apple_store_data[0])
    return apps_ingles

apple_store_data_ingles = filtro_apps_ingles(apple_store_data, 0)
apple_store_data_ingles

[['id',
  'track_name',
  'size_bytes',
  'currency',
  'price',
  'rating_count_tot',
  'rating_count_ver',
  'user_rating',
  'user_rating_ver',
  'ver',
  'cont_rating',
  'prime_genre',
  'sup_devices.num',
  'ipadSc_urls.num',
  'lang.num',
  'vpp_lic'],
 ['284882215',
  'Facebook',
  '389879808',
  'USD',
  '0.0',
  '2974676',
  '212',
  '3.5',
  '3.5',
  '95.0',
  '4+',
  'Social Networking',
  '37',
  '1',
  '29',
  '1'],
 ['389801252',
  'Instagram',
  '113954816',
  'USD',
  '0.0',
  '2161558',
  '1289',
  '4.5',
  '4.0',
  '10.23',
  '12+',
  'Photo & Video',
  '37',
  '0',
  '29',
  '1'],
 ['529479190',
  'Clash of Clans',
  '116476928',
  'USD',
  '0.0',
  '2130805',
  '579',
  '4.5',
  '4.5',
  '9.24.12',
  '9+',
  'Games',
  '38',
  '5',
  '18',
  '1'],
 ['420009108',
  'Temple Run',
  '65921024',
  'USD',
  '0.0',
  '1724546',
  '3842',
  '4.5',
  '4.0',
  '1.6.2',
  '9+',
  'Games',
  '40',
  '5',
  '1',
  '1'],
 ['284035177',
  'Pandora - Music & Radio',
  '130242560'

7. Loop through each dataset to isolate the free apps in separate lists. Make sure you identify the columns describing the app price correctly. Be careful, prices come up as strings ('0', $0.99, $2.99, etc.), so make sure you're not checking an integer or a float in your conditional statements. How many apps are free?


In [47]:
#para google
def get_free_apps(google_play_ingles, price_index):
    free_apps = []

    for row in google_play_ingles[1:]:
        price = row[price_index]

        if price == '0' or price == '0.0':  
            free_apps.append(row)

    return free_apps

google_play_price_index = google_play_ingles[0].index('Price')  
google_play_free_apps = get_free_apps(google_play_ingles, google_play_price_index)
google_play_free_apps



[['Photo Editor & Candy Camera & Grid & ScrapBook',
  'ART_AND_DESIGN',
  '4.1',
  '159',
  '19M',
  '10,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design',
  'January 7, 2018',
  '1.0.0',
  '4.0.3 and up'],
 ['Coloring book moana',
  'ART_AND_DESIGN',
  '3.9',
  '967',
  '14M',
  '500,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design;Pretend Play',
  'January 15, 2018',
  '2.0.0',
  '4.0.3 and up'],
 ['U Launcher Lite – FREE Live Cool Themes, Hide Apps',
  'ART_AND_DESIGN',
  '4.7',
  '87510',
  '8.7M',
  '5,000,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design',
  'August 1, 2018',
  '1.2.4',
  '4.0.3 and up'],
 ['Sketch - Draw & Paint',
  'ART_AND_DESIGN',
  '4.5',
  '215644',
  '25M',
  '50,000,000+',
  'Free',
  '0',
  'Teen',
  'Art & Design',
  'June 8, 2018',
  'Varies with device',
  '4.2 and up'],
 ['Pixel Draw - Number Art Coloring Book',
  'ART_AND_DESIGN',
  '4.3',
  '967',
  '2.8M',
  '100,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design;Creativity',
  'J

In [46]:
#para apple
def get_free_apps(apple_store_data_ingles, price_index):
    free_apps = []

    for row in apple_store_data_ingles[1:]:
        price = row[price_index]

        if price == '0' or price == '0.0':  
            free_apps.append(row)

    return free_apps

apple_store_price_index = apple_store_data_ingles[0].index('price')  
apple_store_free_apps = get_free_apps(apple_store_data_ingles, apple_store_price_index)
apple_store_free_apps

[['284882215',
  'Facebook',
  '389879808',
  'USD',
  '0.0',
  '2974676',
  '212',
  '3.5',
  '3.5',
  '95.0',
  '4+',
  'Social Networking',
  '37',
  '1',
  '29',
  '1'],
 ['389801252',
  'Instagram',
  '113954816',
  'USD',
  '0.0',
  '2161558',
  '1289',
  '4.5',
  '4.0',
  '10.23',
  '12+',
  'Photo & Video',
  '37',
  '0',
  '29',
  '1'],
 ['529479190',
  'Clash of Clans',
  '116476928',
  'USD',
  '0.0',
  '2130805',
  '579',
  '4.5',
  '4.5',
  '9.24.12',
  '9+',
  'Games',
  '38',
  '5',
  '18',
  '1'],
 ['420009108',
  'Temple Run',
  '65921024',
  'USD',
  '0.0',
  '1724546',
  '3842',
  '4.5',
  '4.0',
  '1.6.2',
  '9+',
  'Games',
  '40',
  '5',
  '1',
  '1'],
 ['284035177',
  'Pandora - Music & Radio',
  '130242560',
  'USD',
  '0.0',
  '1126879',
  '3594',
  '4.0',
  '4.5',
  '8.4.1',
  '12+',
  'Music',
  '37',
  '4',
  '1',
  '1'],
 ['429047995',
  'Pinterest',
  '74778624',
  'USD',
  '0.0',
  '1061624',
  '1814',
  '4.5',
  '4.0',
  '6.26',
  '12+',
  'Social Networ

## Exploring data

 8. Build a frequency table for the ```prime_genre``` column of the App Store dataset, and the ```Genres``` and ```Category``` columns of the Google Play data set. What are the most common genres for each market? You must develop and deploy your function

In [53]:
# Solution
from collections import defaultdict

def generate_frequency_table(apple_store_data, column_index):
    frequency_table = defaultdict(int)

    for row in apple_store_data[1:]:
        value = row[column_index]
        frequency_table[value] += 1

    return frequency_table

def get_most_common_genre(frequency_table):
    max_frequency = max(frequency_table.values())
    most_common_genres = [key for key, value in frequency_table.items() if value == max_frequency]

    return most_common_genres, max_frequency

apple_store_genre_index = apple_store_data[0].index('prime_genre')
apple_store_frequency_table = generate_frequency_table(apple_store_data, apple_store_genre_index)
apple_store_most_common_genres, apple_store_max_frequency = get_most_common_genre(apple_store_frequency_table)

apple_store_frequency_table
apple_store_most_common_genres, apple_store_max_frequency

(['Games'], 3862)

In [58]:
#para google
def generate_frequency_table(google_play_ingles, column_index):
    frequency_table = defaultdict(int)

    for row in google_play_ingles[1:]:
        value = row[column_index]
        frequency_table[value] += 1

    return frequency_table

def get_most_common_genre(frequency_table):
    max_frequency = max(frequency_table.values())
    most_common_genres = [key for key, value in frequency_table.items() if value == max_frequency]

    return most_common_genres, max_frequency

google_play_genre_index = google_play_ingles[0].index('Genres')
google_play_frequency_table = generate_frequency_table(google_play_ingles, google_play_genre_index)
google_play_most_common_genres, google_play_max_frequency = get_most_common_genre(google_play_frequency_table)

# Generating frequency table for Category column in Google Play Store dataset
google_play_category_index = google_play_ingles[0].index('Category')
google_play_category_frequency_table = generate_frequency_table(google_play_ingles, google_play_category_index)
google_play_most_common_categories, google_play_max_frequency = get_most_common_genre(google_play_category_frequency_table)

google_play_category_frequency_table
print(google_play_most_common_categories, google_play_max_frequency)
print(google_play_most_common_genres, google_play_max_frequency)

['FAMILY'] 1816
['Tools'] 1816


### El genero más visto en apple es juegos y el google es familia (herramientas) 

 9. One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. Based on that exercise, give one app profile recommendation for Google Play.
 
    For the Google Play data set, we can find the required information in the ```Installs``` column. Then, compute the average number of installs by genre. Hint 1: consider only unique values of the ```Category``` column. Hint 2: If an app has, for instance, ```1,000,000+``` assume it as ```1,000,000``` installs. Hint 3: Be careful with a skewed distribution

In [60]:
# Solution
def calculate_average_installs_by_genre(nuevo_google_play_data, category_index, installs_index):
    genre_installs = {}

    for row in nuevo_google_play_data[1:]:
        category = row[category_index]
        installs = row[installs_index].replace('+', '').replace(',', '')  # Quitamos los + y comas de la columna install 
        installs = float(installs)  # Pasamos a formato float

        if category in genre_installs:
            genre_installs[category][0] += installs
            genre_installs[category][1] += 1
        else:
            genre_installs[category] = [installs, 1]

    average_installs_by_genre = {category: genre_installs[category][0] / genre_installs[category][1] for category in genre_installs}
    return average_installs_by_genre

# Calcular la media de instalaciones por genero en Google Play Store dataset
google_play_category_index = nuevo_google_play_data[0].index('Category')
google_play_installs_index = nuevo_google_play_data[0].index('Installs')
average_installs_by_genre = calculate_average_installs_by_genre(nuevo_google_play_data, google_play_category_index, google_play_installs_index)

# Ordenamos de mayor a menos 
sorted_average_installs = sorted(average_installs_by_genre.items(), key=lambda x: x[1], reverse=True)
sorted_average_installs

[('COMMUNICATION', 35042146.82857143),
 ('VIDEO_PLAYERS', 24091427.73006135),
 ('SOCIAL', 22961790.384937238),
 ('ENTERTAINMENT', 20722156.8627451),
 ('PHOTOGRAPHY', 16545009.448398577),
 ('PRODUCTIVITY', 15489549.114973262),
 ('GAME', 14472288.232533889),
 ('TRAVEL_AND_LOCAL', 13218662.767123288),
 ('TOOLS', 9675661.324062878),
 ('NEWS_AND_MAGAZINES', 9327628.976377953),
 ('BOOKS_AND_REFERENCE', 7504367.459459459),
 ('SHOPPING', 6932419.727722772),
 ('WEATHER', 4570892.658227848),
 ('PERSONALIZATION', 4086386.0853333334),
 ('HEALTH_AND_FITNESS', 3972300.388888889),
 ('MAPS_AND_NAVIGATION', 3841846.4885496185),
 ('SPORTS', 3373767.6861538463),
 ('EDUCATION', 2965983.193277311),
 ('FAMILY', 2416998.638100437),
 ('FOOD_AND_DRINK', 1891060.2767857143),
 ('ART_AND_DESIGN', 1786532.8125),
 ('BUSINESS', 1659916.3452380951),
 ('LIFESTYLE', 1365375.4444444445),
 ('FINANCE', 1319851.4028985507),
 ('HOUSE_AND_HOME', 1313681.9054054054),
 ('DATING', 824129.2807017544),
 ('COMICS', 803234.82142857

## Recomendamos invertir más en apps de comunicación y promover apps medicas por el beneficio para la población.

 10. The App Store dataset doesn´t have the ```installs``` information. Then, you will construct the average rating from the ```rating_count_tot``` column per genre. Based on that exercise, give one app profile recommendation for the App Store

In [61]:
# Solution
def calculate_average_rating_by_genre(apple_store_data, genre_index, rating_count_index):
    genre_ratings = {}

    for row in apple_store_data[1:]:
        genre = row[genre_index]
        rating_count = float(row[rating_count_index])  # Convert rating count to float

        if genre in genre_ratings:
            genre_ratings[genre][0] += rating_count
            genre_ratings[genre][1] += 1
        else:
            genre_ratings[genre] = [rating_count, 1]

    average_rating_by_genre = {genre: genre_ratings[genre][0] / genre_ratings[genre][1] for genre in genre_ratings}
    return average_rating_by_genre

# Calculate average rating by genre (prime_genre column) in the App Store dataset
app_store_genre_index = apple_store_data[0].index('prime_genre')
app_store_rating_count_index = apple_store_data[0].index('rating_count_tot')
average_rating_by_genre = calculate_average_rating_by_genre(apple_store_data, app_store_genre_index, app_store_rating_count_index)

# Sort the average rating by genre in descending order
sorted_average_ratings = sorted(average_rating_by_genre.items(), key=lambda x: x[1], reverse=True)
sorted_average_ratings

[('Social Networking', 45498.89820359281),
 ('Music', 28842.021739130436),
 ('Reference', 22410.84375),
 ('Weather', 22181.027777777777),
 ('Shopping', 18615.32786885246),
 ('Photo & Video', 14352.280802292264),
 ('Travel', 14129.444444444445),
 ('Sports', 14026.929824561403),
 ('Food & Drink', 13938.619047619048),
 ('Games', 13691.996633868463),
 ('News', 13015.066666666668),
 ('Navigation', 11853.95652173913),
 ('Finance', 11047.653846153846),
 ('Health & Fitness', 9913.172222222222),
 ('Productivity', 8051.3258426966295),
 ('Entertainment', 7533.678504672897),
 ('Utilities', 6863.822580645161),
 ('Lifestyle', 6161.763888888889),
 ('Book', 5125.4375),
 ('Business', 4788.087719298245),
 ('Education', 2239.2295805739514),
 ('Catalogs', 1732.5),
 ('Medical', 592.7826086956521)]

## Recomendamos descargar aplicaciones de networking y preocuparse por las aplicaciones medicas